frontend/apps/filemanager/filemanager.lua

```diff
@@ -315,8 +319,26 @@ function FileManager:init()
                         UIManager:close(self.file_dialog)
                     end,
                 },
-            },
-        }
+                {
+                    text_func = function()
+                        if require("readcollection"):checkItemExist(file, "favorites") then
+                            return _("Remove from favorites")
+                        else
+                            return _("Add to favorites")
+                        end
+                    end,
+                    enabled = DocumentRegistry:getProviders(file) ~= nil,
+                    callback = function()
+                        if require("readcollection"):checkItemExist(file, "favorites") then
```

Comment:
While the results of `require` are cached, it should still be preferable to do `local ReadCollection = require(…)`. (And it should be fine at the top of the file I figure?)